tests/integration_tests/datasets/api_tests.py

```diff
@@ -1711,12 +1712,15 @@ def test_bulk_delete_dataset_items(self):
         assert rv.status_code == 200
         expected_response = {"message": f"Deleted {len(datasets)} datasets"}
         assert data == expected_response
+        deleted_datasets = datasets
         datasets = (
             db.session.query(SqlaTable)
             .filter(SqlaTable.table_name.in_(self.fixture_tables_names))
             .all()
         )
         assert datasets == []
+        for dataset in deleted_datasets:
+            setattr(dataset, "_deleted", True)
```

Comment:
This approach doesn't see right, i.e., we shouldn't need to be setting/getting hidden attributes. If I were you, I would re-evaluate the failing tests and see what the issue is. Maybe the results weren't flushed or committed?

Reply:
Updated with a better method of checking whether the object was deleted